packages/mui-base/src/Menu/RadioItem/MenuRadioItem.tsx

```diff
@@ -0,0 +1,219 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { FloatingEvents, useFloatingTree, useListItem } from '@floating-ui/react';
+import { useMenuRadioItem } from './useMenuRadioItem';
+import { useMenuRootContext } from '../Root/MenuRootContext';
+import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import { CustomStyleHookMapping } from '../../utils/getStyleHookProps';
+import { useId } from '../../utils/useId';
+import type { BaseUIComponentProps, GenericHTMLProps } from '../../utils/types';
+import { useForkRef } from '../../utils/useForkRef';
+import { useMenuRadioGroupContext } from '../RadioGroup/MenuRadioGroupContext';
+import { MenuRadioItemContext } from './MenuRadioItemContext';
+
+const customStyleHookMapping: CustomStyleHookMapping<MenuRadioItem.OwnerState> = {
+  checked: (value: boolean) => ({ 'data-state': value ? 'checked' : 'unchecked' }),
```

Comment:
`data-radioitem`? Side note: `data-radio-item` instead? Checkbox had `data-checkboxitem`

Reply:
Both `radio-item` and `radioitem` are fine. I'd be consistent and have no hyphens (this also aligns with ARIA roles)